packages/redux/src/contents/utils.ts

```diff
@@ -20,6 +21,153 @@ export const ENVIRONMENT_CODES = {
+ *
+ * @param {object} metadata - Metadata object with custom attributes.
+ * @returns {number} Ranking number for a specific commerce page.
+ *
+ * @example
+ * const pageRanking = getPageRanking({ custom: { type: "listing" } });
+ * Result of pageRanking === 0;
+ */
+export const getPageRanking = (metadata: Metadata): number => {
+  let ranking = 0;
+  const { gender, brand, category, priceType, id } = metadata.custom;
+
+  // Return default ranking number when no data was returned from each commerce property.
+  // We can't only check if it's empty from 'metadata.custom' because it could have other properties inside.
+  if (!gender && !brand && !category && !priceType && !id) {
+    return ranking;
+  }
+
+  const rankingGender = 10;
+  const rankingBrand = 100;
+  const rankingCategory = 10000;
+  const rankingPriceType = 1000000000;
+  const rankingId = 10000000000;
+
+  Object.entries(metadata.custom).forEach(entry => {
+    const [key, value] = entry;
+
+    switch (key) {
+      case 'gender':
+        !isEmpty(value) && (ranking += rankingGender);
+        break;
+      case 'brand':
+        !isEmpty(value) && (ranking += rankingBrand);
+        break;
+      case 'category':
+        !isEmpty(value) && (ranking += rankingCategory);
+        break;
+      case 'priceType':
+        !isEmpty(value) && (ranking += rankingPriceType);
+        break;
+      case 'id':
+        !isEmpty(value) && (ranking += rankingId);
+        break;
+
+      default:
+        return ranking;
+    }
+  });
+
+  return ranking;
+};
+
+/**
+ * Method to check each page ranking and return the better one.
+ *
+ * @function
+ *
+ * @param {object} result - Commerce page payload result.
+ * @returns {object} Selected page with better ranking.
+ *
+ * @example
+ * const defaultStrategy = getDefaultStrategy({ entries: [...pages] });
+ * Result of defaultStrategy === { entries: [...bestRankedPage] };
+ */
+export const getDefaultStrategy = (
+  result: CommercePagesContentNormalized,
+): CommercePagesContentNormalized => {
+  const bestRankedPage = result.entries
+    .map(page => ({
+      ...page,
+      ranking: getPageRanking(page.metadata),
+    }))
+    .reduce((p, c) => (p.ranking > c.ranking ? p : c));
+
+  return {
+    number: 1,
+    totalPages: 1,
+    totalItems: 1,
+    entries: [bestRankedPage],
+  };
+};
+
+/**
+ * Method to return the main page with all components from other pages.
+ *
+ * @function
+ *
+ * @param {object} result - Commerce page payload result.
+ * @returns {object} Selected page with merged components.
+ *
+ * @example
+ * const mergeStrategy = getMergeStrategy({ entries: [...pages] });
+ * Result of mergeStrategy === { entries: [...mergedPages] };
+ */
+export const getMergeStrategy = (
+  result: CommercePagesContentNormalized,
+): CommercePagesContentNormalized => {
+  // Merge components inside the selected commerce page
+  const mergeResultData = result.entries.reduce((acc, current) => {
+    acc = result.entries[0];
```

Comment:
If you are always overriding the accumulator to a specific value, do you really need a `reduce`? Don't you want some loop just to push new things to the `components` property? 🤔 

Just a question, this made me a bit confused 😖

Reply:
Could be done in both ways, but in fact, I agree with you and it seems more readable now. Can you check?